camel/agents/chat_agent.py

```diff
@@ -271,6 +280,101 @@ def step(
 
         return ChatAgentResponse(output_messages, self.terminated, info)
 
+    def validate_model_response(self, response: Any):
+        if not self.model_backend.stream:
+            if not isinstance(response, dict):
+                raise RuntimeError("OpenAI returned unexpected batch struct")
+        else:
+            if not isinstance(response, GeneratorType):
+                raise RuntimeError("OpenAI returned unexpected stream struct")
+
+    def handle_batch_response(self, response: Dict[str,
+                                                   Any]) -> List[BaseMessage]:
+        r"""
+
+        Args:
+            response (dict): Model response.
+
+        Returns:
+            List[ChatMessage]: A tuple of list of output `ChatMessage`.
+        """
+        output_messages: List[BaseMessage] = []
+        for choice in response["choices"]:
+            chat_message = BaseMessage(role_name=self.role_name,
+                                       role_type=self.role_type,
+                                       meta_dict=dict(),
+                                       content=choice["message"]['content'])
+            output_messages.append(chat_message)
+        return output_messages
+
+    def handle_stream_response(
+        self,
+        response: Any,
+        num_prompt_tokens: int,
+    ) -> Tuple[List[BaseMessage], List[str], Dict[str, int], str]:
+        r"""
+
+        Args:
+            response (dict): Model response.
+            num_prompt_tokens (int): Number of input prompt tokens.
+
+        Returns:
+            tuple: A tuple of list of output `ChatMessage`, list of
+                finish reasons, usage dictionary, and response id.
+        """
+        content_dict = defaultdict(lambda: "")
+        finish_reasons = defaultdict(lambda: "")
+        output_messages: List[BaseMessage] = []
+        response_id: str = ""
+        # All choices in one response share one role
+        role: str = ""
+        for i, chunk in enumerate(response):
```

Comment:
```suggestion
        for chunk in response:
```
Do we use `i` anywhere?